pkg/backup/backup_job.go

```diff
@@ -156,11 +156,23 @@ func backup(
 	evalCtx := execCtx.ExtendedEvalContext()
 	dsp := execCtx.DistSQLPlanner()
 
+	locFilter := sql.SingleLocalityFilter(details.ExecutionLocality)
+	if details.StrictLocalityFiltering {
+		locFilter = make([]roachpb.Locality, 0, len(details.URIsByLocalityKV))
```

Comment:
Isn't this silently clobbering the execution locality? 

Seems like either we should ban specifying both or we should handle both. The latter seems legit, right? you could say you only want to run on `pool=bgwork` within each region, but also say each region has to backup to its regional bucket, right?

Reply:
the ladder would require much more work. For v1, i will ban passing both